numba/cuda/tests/cudapy/test_gufunc.py

```diff
@@ -183,6 +186,49 @@ def copy2d(A, B):
         copy2d(A, out=B)
         self.assertTrue(np.allclose(A, B))
 
+    # Test inefficient use of the GPU where the inputs are all mapped onto a
+    # single thread in a single block.
+    def test_inefficient_guvectorize(self):
+        @guvectorize(['void(float32[:], float32[:], float32[:])'],
+                     '(n),(n)->(n)', target='cuda')
+        def numba_dist_cuda(a, b, dist):
+            len = a.shape[0]
+            for i in range(len):
+                dist[i] = a[i] * b[i]
+
+        a = np.random.rand(1024 * 1024 * 32).astype('float32')
+        b = np.random.rand(1024 * 1024 * 32).astype('float32')
```

Comment:
Looking at the performance of the kernel test led me to also look into how long the guvectorize tests take - I think the data sizes for this and the test below are too large for unit tests, and they take about 5 seconds to run - can we reduce the size of the data please - e.g.:

```suggestion
        a = np.random.rand(1024 * 32).astype('float32')
        b = np.random.rand(1024 * 32).astype('float32')
```